### PR TITLE
Rustify extract filter

### PR DESCRIPTION
**Pull request by rnestler**
This is not yet finished, but I already have some questions :slightly_smiling_face:

### CODE REVIEW

File: test/utils.cpp [hunk 1]
```diff
@@ -373,6 +373,21 @@ TEST_CASE("extract_filter()", "[utils]")
 
 	REQUIRE(filter == "~/bin/script.sh");
 	REQUIRE(url == "https://newsboat.org");
+
+	utils::extract_filter("https://newsboat.org", filter, url);
+
+	REQUIRE(filter == "//newsboat.org");
+	REQUIRE(url == "https://newsboat.org");
```
Comment (rnestler) [inline test/utils.cpp]: The current behavior is quite "interesting" for some edge cases :slightly_smiling_face:. It doesn't really matter, since it is only called on strings which satisfy `is_filter`.
The question is: Do we want to match the current behavior or if it is OK if the Rust version behaves differently?

File: rust/libnewsboat/src/utils.rs [hunk 2]
```diff
@@ -360,6 +360,32 @@ pub fn get_command_output(cmd: &str) -> String {
...
+/// ```
+pub fn extract_filter(line: &str, filter: &mut String, url: &mut String) {
+    let pos = line.find(':').map(|p| p + 1);
+    let pos1 = pos.and_then(|p| line[p..].find(':').map(|p1| p1 + p));
+    println!("{}", line);
+    println!("{:?} {:?}", pos, pos1);
+    match (pos, pos1) {
+        (Some(pos), Some(pos1)) => {
+            *filter = String::from(&line[pos..pos1]);
+            *url = String::from(&line[pos1 + 1..]);
```
Comment (rnestler) [inline rust/libnewsboat/src/utils.rs]: This will panic if the `line` is something with an empty URL like `filter:foo:`. Should we handle that edge case?

**Comment by Minoru** [comment]
Hi, sorry, @pickfire beat you to it with #758. You're asking good questions, though, I'll roll them both into that other PR.

File: test/utils.cpp (hunk 1 above)
Comment (Minoru) [reply test/utils.cpp]: I think it's enough to match the behaviour for cases where we know that input starts with "filter:". In other words, assume `utils::is_filter_url(input) == true`.

File: rust/libnewsboat/src/utils.rs (hunk 2 above)
Comment (Minoru) [reply rust/libnewsboat/src/utils.rs]: Actually, I'd pull this out into separate issue, because #758 takes long enough as it is—let's not add to it.

Comment (rnestler) [reply rust/libnewsboat/src/utils.rs]: Nevermind. I found a much simpler solution with `splitn` which works quite nice.

Comment (Minoru) [reply rust/libnewsboat/src/utils.rs]: I'd like to at least write a test for it once #758 is merged, so I filed #764.

**Comment by rnestler** [comment]
> Hi, sorry, @pickfire beat you to it with #758. You're asking good questions, though, I'll roll them both into that other PR.

Ah dang didn't notice.

**Comment by pickfire** [comment]
@rnestler Sorry, I didn't notice your patch.

File: test/utils.cpp (hunk 1 above)
Comment (pickfire) [reply test/utils.cpp]: This is invalid.
